tests/cases/parser/haltCompiler4.php.tree

```diff
@@ -27,58 +27,16 @@
                         "kind": "CloseParenToken",
                         "textLength": 0
                     },
-                    "semicolon": {
+                    "semicolonOrCloseTag": {
                         "error": "MissingToken",
                         "kind": "SemicolonToken",
                         "textLength": 0
                     },
-                    "data": null
-                }
-            },
-            {
-                "error": "SkippedToken",
-                "kind": "ColonColonToken",
```

Comment:
The reasoning in updating the tests I'd added in the previous PR is that HaltCompiler should halt the compiler and not parse any more statements, so I'd treat the snippet like `__halt_compiler[();]::foo();` with inline html of `::foo();\n`
(this token can't be used in other statements/expressions and is a syntax error)

```
<?php
// invalid
__halt_compiler::foo();
```

The MissingToken in the original tree were due to token_get_all() returning at most 3 tokens after T_HALT_COMPILER then returning T_INLINE_HTML for the rest